src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -45,16 +45,47 @@ namespace dpdk {
     return std::bitset<CHAR_BIT * sizeof(n)>{n}.count();
   }
 
+  static int xdigit2val(unsigned char c)
+  {
+    int val;
+
+    if (isdigit(c))
+      val = c - '0';
+    else if (isupper(c))
+      val = c - 'A' + 10;
+    else
+      val = c - 'a' + 10;
+    return val;
+  }
+
+  static int coremask_bitcount(const char *coremask)
+  {
+    int count = 0;
+
+    if (coremask[0] == '0' && ((coremask[1] == 'x')
+        || (coremask[1] == 'X')))
+      coremask += 2;
+
+    for (int i = 0; coremask[i] != '\0'; i++) {
+      char c = coremask[i];
+      if (isxdigit(c) == 0)
+        return 0;
```

Comment:
i think we need to return `-EINVAL` in this case.

Reply:
The value it returns will be used to campare with `c->_conf->ms_async_op_threads` which is `unsigned`, considering that the returned value is only used in assert, so we choose to return 0 there. Is there any better suggestion?

Reply:
@uniqueyehu we can either continue using `ceph_assert()` to fail the daemon when running into invalid settings. or put something like

```c++
if (int bitcount = coremask_bitcount(coremask.c_str()); bitcount <= 0) {
  return -EINVAL;
} else if (static_cast<uint64_t>(bitcount) > c->_conf->ms_async_op_threads) {
  return -EINVAL;
}
```